cmd/sovereignnode/chainSimulator/tests/bridge/bridge_test.go

```diff
@@ -87,4 +123,34 @@ func TestSovereignChainSimulator_DeployBridgeContractsThenIssueAndDeposit(t *tes
 	require.Nil(t, err)
 	require.NotNil(t, tokenSupply)
 	require.Equal(t, amountToDeposit.String(), tokenSupply.Burned)
+
+	// Wait for outgoing operations to get unconfirmed and check we have one, which is also saved in storage
+	time.Sleep(time.Second)
+
+	outGoingOps := nodeHandler.GetRunTypeComponents().OutGoingOperationsPoolHandler().GetUnconfirmedOperations()
+	require.Len(t, outGoingOps, 1)
+	require.Len(t, outGoingOps[0].OutGoingOperations, 1)
+
+	outGoingOp := outGoingOps[0].OutGoingOperations[0]
+	savedMarshalledTx, err := nodeHandler.GetDataComponents().StorageService().Get(dataRetriever.TransactionUnit, outGoingOp.Hash)
+	require.Nil(t, err)
+	require.NotNil(t, savedMarshalledTx)
+
+	savedTx := &transaction.Transaction{}
+	err = nodeHandler.GetCoreComponents().InternalMarshalizer().Unmarshal(savedTx, savedMarshalledTx)
+	require.Nil(t, err)
+
+	expectedSavedTx := &transaction.Transaction{
+		GasPrice: nodeHandler.GetCoreComponents().EconomicsData().MinGasPrice(),
+		GasLimit: nodeHandler.GetCoreComponents().EconomicsData().ComputeGasLimit(
+			&transaction.Transaction{
+				Data: outGoingOp.Data,
+			}),
+		Data: outGoingOp.Data,
+	}
+	require.Equal(t, expectedSavedTx, savedTx)
+
+	// Generate extra blocks after outgoing operations are created
+	err = cs.GenerateBlocks(10)
+	require.Nil(t, err)
```

Comment:
Maybe a TODO for later to add IncomingHeader with confirmation and then check UnconfirmedOperations == 0. This way we'll have a full cross-chain transaction test.

Reply:
Yes, could be done in another PR